frontend/src/pages/MyData/MyDataStarred.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useDispatch, useSelector } from 'react-redux';
+import { SingleMetaDataResult } from '../../sharedComponents/Metadata/SingleMetaDataResult';
+import { alertActions } from '../../state/actions/alert';
+import { NoResult } from '../MetadataByMunicipality/NoResult';
+
+const MyDataStarredBodyWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  color: #272727;
+  & h2 {
+    margin: 0.5rem 0;
+  }
+`;
+
+const MyDataStarred = () => {
+  const [starredData, setStarredData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const dispatch = useDispatch();
+  const userSelector = useSelector((state) => state.user);
+  const { user: { token } } = userSelector;
+
+  useEffect(() => {
+    const internal = async () => {
+      setLoading(true);
+      try {
+        // TODO: Fetch with a dedicated API call
+        const likedSet = [];
+        const res = await fetch('/api/Metadata');
+        if (res.status === 200) {
+          const receivedMetadataSet = await res.json();
+          receivedMetadataSet.forEach((m, i) => {
+            fetch(`/api/Metadata/${m.uuid}/like`, {
+              method: 'GET',
+              headers: {
+                Authorization: `bearer ${token}`,
+              },
+            }).then((l) => {
+              Promise.resolve(l).then((value) => {
+                value.json().then(({ liked }) => {
+                  if (liked) likedSet.push(m);
+                  setTimeout(() => { // Combating race condition
+                    if (i + 1 === receivedMetadataSet.length) {
+                      setStarredData(likedSet);
+                    }
+                  }, 100);
+                });
+              });
+            });
+          });
+        }
```

Comment:
Two things: Firstly, can we please keep our async stuff to await and promises?
Second: error status (below) doesn't exist if you don't throw an error with that status.
```suggestion
        const res = await fetch('/api/Metadata');
        if (res.status === 200) {
          const receivedMetadataSet = await res.json();
          const likeResPromises = receivedMetadataSet.map(async (m) => {
            const r = await fetch(`/api/Metadata/${m.uuid}/like`, {
              method: 'GET',
              headers: {
                Authorization: `bearer ${token}`,
              },
            });
            return r.json();
          });
          const likeResponses = await Promise.all(likeResPromises);
          const likedMetadata = receivedMetadataSet.filter((_, i) => likeResponses[i].liked);
          setStarredData(likedMetadata);
        } else {
          const err = new Error();
          err.status = res.status;
          throw err;
        }
```